### DISCUSSION TITLE
NextJS tutorial: getStaticProps running but not returning data to home component

### DISCUSSION THREAD

**Discussion by grantoz** [vercel/next.js#19988] [Help] [answered]
Hello! 👋 

I'm following the NextJS tutorial, and I'm a bit stumped at the "Implement getStaticProps" section.

getStaticProps is running, and returning the sample markdown blog data, but the props are not being passed to the <Home> component. 

The console.log in getStaticProps has data, but the console.log log of allPostsData in Home emits undefined:

```javascript
import Head from 'next/head'
import Layout, { siteTitle } from '../components/layout'
import utilStyles from '../styles/utils.module.css'
import { getSortedPostsData } from '../lib/posts'

export default function Home({ allPostsData }) {
  console.log('home', allPostsData)
  return (
    <Layout home>
      <Head>
        <title>{siteTitle}</title>
      </Head>
      <section className={utilStyles.headingMd}>
        <p>Music and code guy.</p>
      </section>
      <section className={`${utilStyles.headingMd} ${utilStyles.padding1px}`}>
        <h2 className={utilStyles.headingLg}>Blog</h2>
        <ul className={utilStyles.list}>
          {/* omitted map over allPostsData here as weirdly it does not exist */}
        </ul>
      </section>
    </Layout>
  )
}
export async function getStaticProps() {
  const allPostsData = getSortedPostsData()
  console.log('getStaticProps', allPostsData)
  return {
    props: {
      allPostsData
    }
  }
}
```

I'd be grateful if someone could help me understand what I've done wrong!    My code so far has been pushed to https://github.com/grantoz/nextjs-blog

Thanks,
Grant

**Comment by jamesmosier** [comment, marked as answer]
Hi @grantoz.

It looks like there is a typo in `_app.js` [here](https://github.com/grantoz/nextjs-blog/blob/bc3100d45eabe40bf878b296a1f1c89ccdeea843/pages/_app.js#L3-L4) with `pageProps`. 

`pageProps` should be a lowercase `p` and in your example it is `PageProps` which is undefined.

`pageProps` are exactly what they sound like, the props from your page. So that would explain why the props were not being sent to your `Home` component.

[See the docs](https://nextjs.org/docs/advanced-features/custom-app) for some more info on the `_app.js` file.

**Reply by grantoz** [reply]
Amazing, thank you so much James!  I usually try to type, rather than copy-paste tutorial code, so as to try to help it embed in my brain - but of course this leaves my work vulnerable to typos.

Thanks once again for your kindness in taking the time to look at my repo and find this.  Much appreciated!

Cheers,
Grant 😄
